Source/SelfService/Web/microservice/dataTable.tsx

```diff
@@ -0,0 +1,211 @@
+
+const styles = {
+    dataTable: {
+        'background': 'linear-gradient(180deg, rgba(255, 255, 255, 0.05) 0%, rgba(255, 255, 255, 0.05) 100%), #191A21;',
+        '.MuiDataGrid-row': {
+            cursor: 'pointer'
+        },
+        '.MuiDataGrid-columnSeparator': {
+            display: 'none'
+        },
+        '.MuiDataGrid-columnHeader:focus': {
+            outline: 'none'
+        },
+        '.MuiDataGrid-columnHeader:focus-within': {
+            outline: 'none'
+        },
+        '.MuiDataGrid-columnHeaderDraggableContainer': {
+            outline: 'none'
+        },
+        '.MuiDataGrid-cell:focus-within': {
+            outline: 'none'
+        },
+        '.MuiIconButton-root': {
+            'visibility': 'visible',
+            ':hover': {
+                backgroundColor: 'transparent'
+            }
+        },
+        '.MuiDataGrid-sortIcon': {
+            color: 'rgba(255, 255, 255, 0.38);',
+            width: '20px',
+            height: '20px'
+        },
+        '.MuiDataGrid-columnHeader:not(.MuiDataGrid-columnHeader--sorted) .MuiDataGrid-sortIcon': {
+            opacity: 1
+        }
+    },
+    status: {
+        display: 'flex',
+        justifyContent: 'center'
+    },
+    statusTitle: {
+        fontWeight: '500',
+        fontSize: '12px',
+        lineHeight: '22px',
+        letterSpacing: '0.06em',
+        textTransform: 'uppercase',
+        marginInlineStart: '10px'
+    }
+};
+
+export type MicroserviceObject = {
+    id: string,
+    name: string,
+    kind: string,
+    environment: string,
+    live: MicroserviceInfo,
+    edit: {},
+    status?: []
+};
+
+type DataTableProps = {
+    environment: string,
+    application: HttpResponseApplication,
+    microservices: MicroserviceObject[]
+};
+
+export const DataTable: React.FC<DataTableProps> = ({ application, environment, microservices }: DataTableProps) => {
+    const history = useHistory();
+    const [rows, setRows] = useState<(MicroserviceObject | undefined)[]>([]);
+
+    useEffect(() => {
+        const mapStatusToMicroservicesAndSetState = async () => {
+            const rows = await Promise.all(microservices.map(async microservice => {
+                const status = await getMicroserviceStatus(microservice.id);
+
+                return {
+                    ...microservice,
+                    status
+                } as MicroserviceObject;
+            }));
+
+            setRows(rows);
+        };
+
+        mapStatusToMicroservicesAndSetState();
+    }, [setRows]);
+
+    const getMicroserviceStatus = (microserviceId: string) => {
+        return new Promise((resolve, reject) => {
+            resolve(
+                getPodStatus(application.id, environment, microserviceId)
+                    .then(res => res.pods)
+                    .catch(err => reject(err))
+            );
+        });
+    };
+
+    const columns: GridColDef[] = [
```

Comment:
These columns don't change - so they could be moved outside the component rendering function.

Reply:
I got 'Public URL' sorting to work and now it need row state (and therefore stay in same place?).